src/edu/stanford/dlss/was/WasapiConnection.java

```diff
@@ -0,0 +1,129 @@
+
+public class WasapiConnection {
+
+    private CloseableHttpClient wasapiClient;
+    private HttpClientContext wasapiContext;
+    private BasicCookieStore cookieStore;
+
+    public WasapiConnection(WasapiDownloaderSettings settings) throws IOException, UnsupportedEncodingException {
+        wasapiClient = initializeClient();
+        HttpPost loginRequest = createLoginRequest(settings);
+        login(loginRequest);
+        wasapiContext = HttpClientContext.create();
+        wasapiContext.setCookieStore(cookieStore);
+    }
+
+    public WasapiResponse jsonQuery(String requestURL) throws IOException {
+        HttpGet jsonRequest = new HttpGet(requestURL);
+
+        ResponseHandler<WasapiResponse> rh = new ResponseHandler<WasapiResponse>() {
+            @Override
+            public WasapiResponse handleResponse(final HttpResponse response) throws IOException {
+                StatusLine statusLine = response.getStatusLine();
+                HttpEntity entity = response.getEntity();
+
+                if(validateResponse(response.getStatusLine(), (entity == null))) {
+                    WasapiResponseParser wasapiParser = new WasapiResponseParser();
+                    return wasapiParser.parse(entity.getContent());
+                }
+                else return null;
+            }
+        };
+        return wasapiClient.execute(jsonRequest, rh, wasapiContext);
+    }
+
+
+    // what should this return?
+    public Boolean downloadQuery(String downloadURL, final String outputFilename) throws ClientProtocolException, HttpResponseException, IOException {
+        HttpGet fileRequest = new HttpGet(downloadURL);
+
+        ResponseHandler<Boolean> rh = new ResponseHandler<Boolean>() {
+            @Override
+            public Boolean handleResponse(final HttpResponse response) throws IOException {
+                StatusLine statusLine = response.getStatusLine();
+                HttpEntity entity = response.getEntity();
+
+                if(validateResponse(response.getStatusLine(), (entity == null))) {
+                    FileOutputStream fouts = new FileOutputStream(outputFilename);
+                    entity.writeTo(fouts);
+                    fouts.close();
+                    return true;
+                }
+                else return false;
+            }
+        };
+        return wasapiClient.execute(fileRequest, rh, wasapiContext);
+    }
+
+
+    public void close() throws IOException {
+        wasapiClient.close();
+    }
+
+
+    private boolean validateResponse(StatusLine statusLine, boolean entityIsNull) throws ClientProtocolException, HttpResponseException {
+        if(statusLine.getStatusCode() >= 300) {
+            throw new HttpResponseException(statusLine.getStatusCode(), statusLine.getReasonPhrase());
+        }
+        if(entityIsNull) {
+            throw new ClientProtocolException("Response contains no content");
+        }
+        return true;
+    }
+
+
```

Comment:
I'm not sure I would put 3 blank lines here, though 2 lines between the public and the private sections works for me.